src/datasets/builder.py

```diff
@@ -627,7 +636,19 @@ def _download_and_prepare(self, dl_manager, verify_infos, **prepare_split_kwargs
         # Generating data for all splits
         split_dict = SplitDict(dataset_name=self.name)
         split_generators_kwargs = self._make_split_generators_kwargs(prepare_split_kwargs)
-        split_generators = self._split_generators(dl_manager, **split_generators_kwargs)
+        try:
+            split_generators = self._split_generators(
+                dl_manager, splits=dl_manager._download_config.splits, **split_generators_kwargs
+            )
+        except TypeError:
+            split_generators = self._split_generators(dl_manager, **split_generators_kwargs)
```

Comment:
You can use `"splits" in inspect.signature(self._split_generators).parameters.keys()` instead of try/catch

Reply:
I just applied the Pythonic EAFP principle: "It is Easier to Ask for Forgiveness than Permission".

If you really think it is better to make 4 additional calls (`inspect.signature`, `keys`, `if`, `in`, besides the import of the `inspect` package), I can refactor the code. 😅

Reply:
I have a small preference for `inspect` since the implementation of _split_generators may raise a TypeError, especially when someone is implementing this method and may have added bugs (but no strong opinion). It would be nice to not catch and hide errors from the users' code or they may have trouble debugging.

Reply:
In the except clause, the same method is called again (the only difference: without passing the parameter `splits`). Therefore, any user-created exception will be raised anyway. 😉

Reply:
Right :)